bin/launch_check_model.py

```diff
@@ -1,29 +1,44 @@
+#!/usr/bin/env python3
+
 import argparse
 import torch.nn as nn
 import torch
 import os
 import src.data.handlertorch as handlertorch
+import json
 
 from copy import deepcopy
 from src.utils.yaml_model_schema import YamlRayConfigLoader
 from launch_utils import import_class_from_file, get_experiment
+from json_schema import JsonSchema
+
+
+
+def get_args():
+
+    """get the arguments when using from the commandline"""
 
-def arg_parser():
     parser = argparse.ArgumentParser(description="Launch check_model")
-    parser.add_argument("--input", type=str, help="Path to input csv file", required=True)
-    parser.add_argument("--model", type=str, help="Path to model file", required=False)
-    parser.add_argument("--experiment", type=str, help="Experiment to run", required=True)
-    parser.add_argument("--config", type=str, help="Path to json config training file", required=False)
-    return parser.parse_args()
-class CheckModelWrapper:
+    parser.add_argument("-d", "--data", type=str, required=True, metavar="FILE", help="Path to input csv file")
+    parser.add_argument("-m", "--model", type=str, required=True, metavar="FILE", help="Path to model file")
+    parser.add_argument("-e", "--experiment", type=str, required=True, metavar="FILE", help="Experiment config file. From this the experiment class name is extracted.")
+    parser.add_argument("-c", "--config", type=str, required=True, metavar="FILE", help="Path to json config training file")
+    
+    args = parser.parse_args()
+    return args
+
+
+class CheckModelWrapper():
+
     def __init__(self, model: nn.Module, config_instance: dict, data_path: str, experiment: object):
         self.model = model(**config_instance["model_params"])
         # get the optimizer from pytorch
-        optimizer = getattr(torch.optim, config_instance["optimizer"]["method"])
+        optimizer = getattr(torch.optim, config_instance["optimizer_params"]["method"])
         # get the loss function from pytorch
         self.loss_fn = getattr(torch.nn, config_instance["loss_params"]["loss_fn"])()
-        # instantiate the optimizer
-        self.optimizer = optimizer(self.model.parameters(), **config_instance["optimizer_params"])
+        # instantiate the optimizer, get all optimizer parameters except the names of the optimizers themselves
+        optimizer_params_values = {key: value for key, value in config_instance["optimizer_params"].items() if key != "method"}
```

Comment:
now reading this, i wonder maybe it is better to have 'optimizer' and 'optimizer_params' separate?

Reply:
indeed each optimizer https://pytorch.org/docs/stable/optim.html have different set of params